browser/background.js

```diff
@@ -45,6 +45,71 @@ addListener('ajax', async ({ method, url, headers, data }) => {
 	};
 });
 
+addListener('authFlow', async ({ domain, clientId, scope, interactive }) => {
+	const url = new URL(domain);
+	url.searchParams.set('client_id', clientId);
+	url.searchParams.set('scope', scope);
+	url.searchParams.set('response_type', 'token');
+
+	let responseUrl;
+
+	if (!interactive) {
+		// Try to get an auth token non-interactively using the native API.
+		// The browser will throw an error if it cannot complete without user input.
+		url.searchParams.set('redirect_uri', chrome.identity.getRedirectURL());
+		// chrome.identity is not implemented in Edge, so this throws.
+		// That's okay, because noninteractive auth is expected to fail sometimes.
+		responseUrl = await apiToPromise(chrome.identity.launchWebAuthFlow)({ url: url.href, interactive: false });
+	} else {
+		// Try to get an auth token interactively.
+		// We do not use the native API, because it requires a redirect to a potentially
+		// confusing domain, like chromiumapp.org or allizom.org.
+		// Instead, open a tab which will redirect to the sensible domain redditenhancementsuite.com
+		// and track it with the chrome.tabs API.
+		const redirectUri = 'https://redditenhancementsuite.com/oauth';
+		url.searchParams.set('redirect_uri', redirectUri);
+
+		const { tabs: [{ id }] } = await apiToPromise(chrome.windows.create)({ url: url.href, type: 'popup', focused: true });
```

Comment:
Firefox 53: `Type error for parameter createData (Property "focused" is unsupported by Firefox) for windows.create.`